src/Microsoft.ML.Tokenizers/Model/EnglishRoberta.cs

```diff
@@ -127,29 +127,30 @@ public EnglishRoberta(Stream vocabularyStream, Stream mergeStream, Stream highes
         /// <summary>
         /// Map the tokenized Id to the token.
         /// </summary>
-        /// <param name="id">The Id to map to the token.</param>
-        /// <param name="skipSpecialTokens">Indicate if want to skip the special tokens during the decoding.</param>
-        /// <returns>The mapped token of the Id.</returns>
-        public override string? IdToToken(int id, bool skipSpecialTokens = false) =>
-            skipSpecialTokens && id < 0 ? null : _vocabReverse.TryGetValue(id, out var value) ? value : null;
-
-        /// <summary>
-        /// Map the tokenized Id to the original string while filtering out unsupported characters.
-        /// </summary>
         /// <param name="id">The Id to map to the string.</param>
         /// <param name="skipSpecialTokens">Indicate if want to skip the special tokens during the decoding.</param>
+        /// <param name="filterUnsupportedChars">Indicate if want to filter the unsupported characters during the decoding.</param>
         /// <returns>The mapped token of the Id.</returns>
-        public string? IdToFilteredToken(int id, bool skipSpecialTokens = false)
+        public override string? IdToToken(int id, bool skipSpecialTokens = false, bool filterUnsupportedChars = true)
         {
             if (skipSpecialTokens && id < 0)
+            {
                 return null;
+            }
+
             if (_vocabReverse.TryGetValue(id, out var value))
             {
-                var textChars = string.Join("", value)
-                    .Where(c => _unicodeToByte.ContainsKey(c))
-                    .Select(c => _unicodeToByte[c]);
-                var text = new string(textChars.ToArray());
-                return text;
+                if (filterUnsupportedChars)
+                {
+                    var textChars = string.Join("", value)
```

Comment:
This is just string.Concat

Reply:
That said, what is the type of value here? (One of the reasons I dislike var is I can't easily see what it is.)

But, assuming it's a string, I don't understand the purpose of this string.Join("", value)... is that just going to return a new string that's identical to value?